staticd/static_routes.c

```diff
@@ -0,0 +1,506 @@
+/*
+ * STATICd - vrf code
+ * Copyright (C) 2018 Cumulus Networks, Inc.
+ *               Donald Sharp
+ *
+ * FRR is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * FRR is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+#include <zebra.h>
+
+#include <lib/nexthop.h>
+#include <lib/memory.h>
+#include <lib/srcdest_table.h>
+#include <lib/if.h>
+#include <lib/vty.h>
+#include <lib/vrf.h>
+#include <lib/memory.h>
+
+#include "static_vrf.h"
+#include "static_routes.h"
+#include "static_memory.h"
+#include "static_zebra.h"
+
+/* Install static route into rib. */
+static void static_install_route(struct route_node *rn)
+{
+	struct static_route *si;
+
+	for (si = rn->info; si; si = si->next)
+		static_zebra_nht_register(si, true);
+
+	si = rn->info;
+	static_zebra_route_add(rn, si->vrf_id, true);
```

Comment:
On my tests staticd segfaults in this line (`si` is `NULL`) if I do this:
```
debian(config)# ip route 50.0.0.0/8 10.0.1.2
debian(config)# ip route 60.0.0.0/8 50.0.0.1
debian(config)#
root@debian:~#
root@debian:~# ip link add vrf-red type vrf table 10
root@debian:~# ip link set dev vrf-red up
root@debian:~# ip link set dev rt1-eth0 master vrf-red
```
Could you check what's the problem here?

Reply:
fixed